### PR TITLE
added functionality to reset password

### LINKED ISSUE #81: Add backend for reset password
**Is your feature request related to a problem? Please describe.**
As of now there is UI for forgot password in the frontend, so we need to have the backend for the fully implementation of that feature.

**Describe the solution you'd like**
A route will be formed through which reset password request will be sent to the backend and then just after that request a token will be generated for that registered email id which will be valid for an hour and after that, the request for updating password will be sent to the backend through another route. 

**Additional context**
**Working on it**

### PR DESCRIPTION
**Pull request by Rupeshiya**
Fixes #81 
Added functionality to reset password
**Here is the screenshots:**
**STEP1:**
pass0
**STEP2:**
pass1

The token during screenshot was 64bytes long, reduced to 32 now (**UPDATED**)

Please review @devesh-verma @vaibhavdaren

### CODE REVIEW

**Comment by devesh-verma** [comment]
On success updation of the password just return status code as 204. No success message.

**Comment by Rupeshiya** [comment]
Done @devesh-verma !

File: app/routes/user.js [hunk 1]
```diff
@@ -23,6 +23,17 @@ router.patch(
   userController.userProfileUpdate
 )
 
+// user forgot password request
+router.post(
+  '/resetPass',
```
Comment (devesh-verma) [inline app/routes/user.js]: routes should be all smallcase

**Comment by devesh-verma** [comment]
@Rupeshiya how come to travis build for this PR? any clue

**Comment by devesh-verma** [comment]
> Done @devesh-verma !
> Why checks failing here! @devesh-verma Is it failing **due to comments** in the code?
> 
> It's **CI test cases** passing locally with the PR and updated code.
> test

Those are the lines which are not executed while the test suite runs.

**Comment by Rupeshiya** [comment]
> @Rupeshiya how come no travis build for this PR? any clue

checking this @devesh-verma !

**Pull request merged**